crates/wac-types/src/component.rs

```diff
@@ -684,9 +789,9 @@ impl fmt::Display for FuncKind {
 #[cfg_attr(feature = "serde", derive(serde::Serialize))]
 #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
 pub struct ResourceAlias {
-    /// The owning interface for the resource.
+    /// The foreign owning interface for the resource.
     ///
-    /// This may be `None` if the resource is owned by a world.
+    /// This may be `None` if the resource does not have a foreign interface owner.
```

Comment:
I'm sure to forget about when there isn't a foreign owner so it'd be nice to document that.

```suggestion
    /// This may be `None` if the resource does not have a foreign interface owner such as in a world or when aliasing within the same interface. 
```